### PR TITLE
Update transformer.md

### PR DESCRIPTION
**Pull request by zhmou**
*Description of changes:*
fix a typo in pytorch implementation about _TransformerDecoderBlock_
torch.cat() takes 2 parameters, the second one's name is "dim", not "axis".
see [pytorch docs about torch.cat()](https://pytorch.org/docs/stable/generated/torch.cat.html)

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Review by AnirudhDagar — APPROVED** [review]
Thanks @zhmou, this is definitely more consistent even though the `axis` arg works as a numpy compatible analogue to `dim`!

Could you please send another PR to add your name to our acknowledgement?
https://github.com/d2l-ai/d2l-en/blob/master/chapter_preface/index.md (Like #2146)
